test/integration/crl_test.go

```diff
@@ -1,3 +1,87 @@
 //go:build integration
 
 package integration
+
+import (
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/letsencrypt/boulder/core"
+	"github.com/letsencrypt/boulder/test"
+)
+
+// runUpdater executes the crl-updater binary with the -runOnce flag, and
+// returns when it completes.
+func runUpdater(t *testing.T) {
+	t.Helper()
+
+	config_dir := os.Getenv("BOULDER_CONFIG_DIR")
+	if config_dir == "" {
+		t.Fatal("couldn't determine configuration directory")
+	}
+
+	bin_path, err := filepath.Abs("bin/crl-updater")
+	test.AssertNotError(t, err, "computing crl-updater path")
+
+	c := exec.Command(bin_path, "-config", path.Join(config_dir, "crl-updater.json"), "-debug-addr", ":8022", "-runOnce")
+	err = c.Run()
+	test.AssertNotError(t, err, "failed to run crl-updater")
+}
+
+// TestCRLPipeline runs an end-to-end test of the crl issuance process, ensuring
+// that the correct number of properly-formed and validly-signed CRLs are sent
+// to our fake S3 service.
+func TestCRLPipeline(t *testing.T) {
+	// The crl-updater and crl-storer are not yet deployed in Prod, so only run
+	// this test in config-next.
+	if !strings.Contains(os.Getenv("BOULDER_CONFIG_DIR"), "config-next") {
+		return
+	}
+
+	// Basic setup.
+	os.Setenv("DIRECTORY", "http://boulder:4001/directory")
+	client, err := makeClient()
+	test.AssertNotError(t, err, "creating acme client")
+
+	// Issue a test certificate and save its serial number.
+	res, err := authAndIssue(client, nil, []string{random_domain()})
+	if err != nil || len(res.certs) < 1 {
+		t.Fatal("failed to create test certificate")
```

Comment:
Nit, you should probably surface the contents of the `err` here.

Reply:
`err` is nil if we hit the `len(res.certs) < 1` case. But honestly I can probably just remove that case entirely; I don't think there's any situation in which there's no error but also no certs.